test/utils/ansible-test/lib/util.py

```diff
@@ -0,0 +1,403 @@
+
+    communicate = False
+
+    if stdin is not None:
+        data = None
+        communicate = True
+    elif data is not None:
+        stdin = subprocess.PIPE
+        communicate = True
+
+    if stdout:
+        communicate = True
+
+    if capture:
+        stdout = stdout or subprocess.PIPE
+        stderr = subprocess.PIPE
+        communicate = True
+    else:
+        stderr = None
+
+    start = time.time()
+
+    try:
+        process = subprocess.Popen(cmd, env=env, stdin=stdin, stdout=stdout, stderr=stderr, cwd=cwd)
+    except OSError as ex:
+        if ex.errno == errno.ENOENT:
+            raise ApplicationError('Required program "%s" not found.' % cmd[0])
+        raise
+
+    if communicate:
+        stdout, stderr = process.communicate(data)
+    else:
+        process.wait()
+        stdout, stderr = None, None
+
+    status = process.returncode
+    runtime = time.time() - start
+
+    display.info('Command exited with status %s after %s seconds.' % (status, runtime), verbosity=4)
+
+    if status == 0:
+        return stdout, stderr
+
+    raise SubprocessError(cmd, status, stdout, stderr, runtime)
+
+
+def common_environment():
+    """Common environment used for executing all programs."""
+    env = dict(
+        LC_ALL='en_US.UTF-8',
+        PATH=os.environ.get('PATH', os.defpath),
+    )
+
+    required = (
+        'HOME',
+    )
+
+    optional = (
+        'HTTPTESTER',
+    )
+
+    env.update(pass_vars(required=required, optional=optional))
+
+    return env
+
+
+def pass_vars(required=None, optional=None):
+    """
+    :type required: collections.Iterable[str]
+    :type optional: collections.Iterable[str]
+    :rtype: dict[str, str]
+    """
+    env = {}
+
+    for name in required:
+        if name not in os.environ:
+            raise MissingEnvironmentVariable(name)
+        env[name] = os.environ[name]
+
+    for name in optional:
+        if name not in os.environ:
+            continue
+        env[name] = os.environ[name]
+
+    return env
+
+
+def deepest_path(path_a, path_b):
+    """Return the deepest of two paths, or None if the paths are unrelated.
+    :type path_a: str
+    :type path_b: str
+    :return: str | None
+    """
+    if path_a == '.':
+        path_a = ''
+
+    if path_b == '.':
+        path_b = ''
+
+    if path_a.startswith(path_b):
+        return path_a or '.'
+
+    if path_b.startswith(path_a):
+        return path_b or '.'
+
+    return None
+
+
+def make_dirs(path):
+    """
+    :type path: str
+    """
+    try:
+        os.makedirs(path)
+    except OSError as ex:
+        if ex.errno != errno.EEXIST:
+            raise
+
+
+class Display(object):
+    """Manages color console output."""
+    clear = '\033[0m'
+    red = '\033[31m'
+    green = '\033[32m'
+    yellow = '\033[33m'
+    blue = '\033[34m'
+    purple = '\033[35m'
+    cyan = '\033[36m'
+
+    verbosity_colors = {
+        0: None,
+        1: green,
+        2: blue,
+        3: cyan,
```

Comment:
There are actually 4 levels of verbosity in ansible-playbook. 4th is often lower level connection info.